libkmod/libkmod-index.c

```diff
@@ -699,7 +699,7 @@ static struct index_mm_node *index_mm_read_node(struct index_mm *idx, uint32_t o
 		first = read_char_mm(&p);
 		last = read_char_mm(&p);
 
-		if (first > last || first < 0 || last < 0)
+		if (first > last || first > SCHAR_MAX || last > SCHAR_MAX)
```

Comment:
While technically correct, changing it to `>= INDEX_CHILDMAX` avoids this very technical description. After all, we have to check if the index into child array is valid. If you adjust it, I just mention here again that it turns the `>`-check into `>=`-check. I've failed spotting this while I later adjusted code due to such requests more than once ... :)